source/loader/linux/ur_loader_lin.cpp

```diff
@@ -0,0 +1,16 @@
+/*
+ *
+ * Copyright (C) 2023 Intel Corporation
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ */
+#include "ur_loader.hpp"
+
+namespace loader {
+
+FUNC_PTR platform_t::getFunctionPtr(const char *func_name) {
+    return dlsym(handle.get(), func_name);
```

Comment:
You should keep all the dynamic shared library abstractions in one place - `ur_lib_loader`.